test_bgp_ipv6_ll_peering.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2023 Nathan Mangar
+
+"""
+Check if IPv6 Link-Local BGP peering works fine.
+"""
+
+__topotests_file__ = "bgp_ipv6_ll_peering/test_bgp_ipv6_ll_peering.py"
+__topotests_gitrev__ = "0f4a09ac25d42601f42d37e044f8630ec7d31507"
+
+# pylint: disable=invalid-name, missing-class-docstring, missing-function-docstring, line-too-long, consider-using-f-string, wildcard-import, unused-wildcard-import, f-string-without-interpolation, too-few-public-methods, unused-argument
+
+from topotato import *
+
+
+@topology_fixture()
+def topology(topo):
+    """
+    [ r1 ]
+      |
+    { s1 }
+      |
+    [ r2 ]
+    """
+    topo.router("r1").iface_to("s1").ip6.append("fe80:1::1/64")
```

Comment:
this is not needed btw, there should always be a link-local address when IPv6 is enabled. check `router("r1").iface_to("s1").ll6`